### ISSUE TITLE
Failed to build, complaining about -std=c++11 is not allowed with 'C'

### ISSUE THREAD

**Issue by rabbull** [ducha-aiki/pyransac#3]
Environment:
 - cmake: 3.16.4
 - make: 4.3
 - clang: 9.0.1
 - python: 3.8.1
 - gfortran: 9.2.1

Makefile generation goes without error nor warning:
```
$ /bin/cmake /path/to/pyransac\
  -DCMAKE_BUILD_TYPE=Debug \
  -DCMAKE_MAKE_PROGRAM=/bin/make \
  -DCMAKE_C_COMPILER=/bin/clang \
  -DCMAKE_CXX_COMPILER=/bin/clang++ \
  -G "CodeBlocks - Unix Makefiles"
-- The C compiler identification is Clang 9.0.1
-- The CXX compiler identification is Clang 9.0.1
-- Check for working C compiler: /bin/clang
-- Check for working C compiler: /bin/clang -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /bin/clang++
-- Check for working CXX compiler: /bin/clang++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found OpenMP_C: -fopenmp=libomp (found version "3.1") 
-- Found OpenMP_CXX: -fopenmp=libomp (found version "3.1") 
-- Found OpenMP: TRUE (found version "3.1")  
-- The Fortran compiler identification is GNU 9.2.1
-- Check for working Fortran compiler: /bin/f95
-- Check for working Fortran compiler: /bin/f95  -- works
-- Detecting Fortran compiler ABI info
-- Detecting Fortran compiler ABI info - done
-- Checking whether /bin/f95 supports Fortran 90
-- Checking whether /bin/f95 supports Fortran 90 -- yes
-- Looking for Fortran sgemm
-- Looking for Fortran sgemm - not found
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Looking for Fortran sgemm
-- Looking for Fortran sgemm - found
-- Found BLAS: /usr/lib/libblas.so  
-- Looking for Fortran cheev
-- Looking for Fortran cheev - not found
-- Looking for Fortran cheev
-- Looking for Fortran cheev - found
-- A library with LAPACK API found.
-- Found PythonInterp: /usr/bin/python (found version "3.8.1") 
-- Found PythonLibs: /usr/lib/libpython3.8.so
-- pybind11 v2.1.1
-- Performing Test HAS_CPP14_FLAG
-- Performing Test HAS_CPP14_FLAG - Success
-- Performing Test HAS_CPP11_FLAG
-- Performing Test HAS_CPP11_FLAG - Success
-- Performing Test HAS_FLTO
-- Performing Test HAS_FLTO - Success
-- LTO enabled
-- Configuring done
-- Generating done
-- Build files have been written to: /path/to/build
```

But make failed:
```
$ make
make[1]: Entering directory '/path/to/pyransac/build'
make[2]: Entering directory '/path/to/pyransac/build'
Scanning dependencies of target matutls
make[2]: Leaving directory '/path/to/pyransac/build'
make[2]: Entering directory '/path/to/pyransac/build'
[  1%] Building C object src/pyransac/matutls/CMakeFiles/matutls.dir/atou1.c.o
error: invalid argument '-std=c++11' not allowed with 'C'
make[2]: *** [src/pyransac/matutls/CMakeFiles/matutls.dir/build.make:63: src/pyransac/matutls/CMakeFiles/matutls.dir/atou1.c.o] Error 1
make[2]: Leaving directory '/path/to/pyransac/build'
make[1]: *** [CMakeFiles/Makefile2:144: src/pyransac/matutls/CMakeFiles/matutls.dir/all] Error 2
make[1]: Leaving directory '/path/to/pyransac/build'
make: *** [Makefile:84: all] Error 2
```

**Comment by rabbull** [comment]
Problem solved by using GCC instead of Clang.

For gcc, using -std=c++xx with C files is only a warning, while for clang it's an error. Maybe change "A modern compiler with C++11 support" in readme to "GCC"?

**Comment by ricshaw** [comment]
I got it to install on my mac with this: 
sudo CC=/usr/local/bin/gcc-9 CXX=/usr/local/bin/g++-9 python3 ./setup.py build

It seems to build fine, but when I try to import it in python it says No module named 'pyransac'. Do you know how to fix this?

**Comment by ducha-aiki** [comment]
Unfortunately I don't have mac. But it seems that you only built the module, not installed it.
try setup.py install instead of setup.py build

**Comment by ricshaw** [comment]
That worked thank you!

**Comment by sailor-z** [comment]
Hi,

I got a similar error in Ubuntu 16.04, cmake 3.17.1, gcc 6.1, but it was successfully compiled in another computer with  Ubuntu 18.04, cmake 3.10.2 , gcc 5.5.
![image](https://user-images.githubusercontent.com/26895204/79092991-4d58a980-7d85-11ea-8535-ac96d34c76ed.png)

**Comment by sailor-z** [comment]
I've solved the problem by replacing gcc 6.1 with gcc 4.8.

**Comment by rabbull** [comment]
As the repo is now moved and this problem is fully discussed (I guess?), this issue should be close I think.
